File: C++/1190. Reverse Substrings Between Each Pair of Parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        string ans="";
        stack<int>st;        
        for(int i=0;i<s.size();i++){
            if(s[i]=='(')
                st.push(i);

            else if(s[i]==')'){
                reverse(s.begin()+st.top(),s.begin()+i);
                st.pop();
            }
        }
        for(int i=0;i<s.size();i++){
            if(s[i]=='('||s[i]==')')
            continue;
                 ans+=s[i];
        }      
        return ans;
    }
};

File: JAVA/1190. Reverse Substrings Between Each Pair of Parentheses.java
class Solution {
     public static void main(String[] args) {
        String s = "(ed(et(oc))el)";
        Solution bp = new Solution();
        bp.reverseParentheses(s);
    }
    public String reverseParentheses(String s) {
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                count++;
            }
        }
        for (int j=0;j<count;j++) {
            int RP = s.indexOf(')');
            int LP = s.lastIndexOf('(', RP);
            String innenString = s.substring(LP + 1, RP);
            String reversedSubstring = new StringBuilder(innenString).reverse().toString();
            String resultString = s.substring(0, LP) + reversedSubstring + s.substring(RP + 1);
            s = resultString;
        }
        return s;
    }
}
